feat: Adding Apache Arrow FlightShim for connector federation in C++ workers (#26369)
prestodb/presto: presto-docs/src/main/sphinx/federation/deployment.rst

====================================================
Deploying FlightShim Server for Connector Federation
====================================================

About FlightShim
----------------

FlightShim is used to support connector federation on a native cluster. An
Apache Arrow Flight server is used to load Java-based Presto connectors that
are not otherwise available to a Prestissimo worker. The native worker can use
the Arrow Federation connector to register federated connectors and will then
forward table read requests to FlightShim after receiving split information
from the coordinator.

Installing FlightShim
---------------------

Download the Presto FlightShim tarball, :maven_download:`flight-shim`, and unpack it.
The tarball will contain a single top-level directory,
|presto_flight_shim_release|, which we will call the *installation* directory.

> FlightShim needs a *data* directory for storing logs, etc.

Reviewer: ```suggestion
FlightShim needs a *data* directory for storing logs and other files.
```
Avoid Latin abbreviations. See the [GitLab documentation style guide recommended word list entry for e.g.](https://docs.gitlab.com/development/documentation/styleguide/word_list/#eg) for discussion and alternatives. 

Note: given line 29, did you mean to refer to the configuration directory `etc` by name? If that is what you want to do here, please format the directory name as `etc`.

We recommend creating a data directory outside of the installation directory,

> which allows it to be easily preserved when upgrading Presto.

Reviewer: ```suggestion
which allows it to be preserved when upgrading Presto.
```

Configuring FlightShim
----------------------

Create an ``etc`` directory inside the installation directory.
Similar to the installation of Presto, this will hold the following configuration:

* JVM Config: command line options for the Java Virtual Machine

> * Config Properties: configuration for the Presto FlightShim server

Reviewer: ```suggestion
* Configuration Properties: configuration for the Presto FlightShim server
```

* Catalog Properties: configuration for :doc:`/connector` (data sources).
  The available catalog configuration properties for a connector are described
  in the respective connector documentation.

.. _jvm_config:

> JVM Config

Reviewer: ```suggestion
JVM Configuration
```

> ^^^^^^^^^^

Reviewer: Please edit this line to match the number of characters in the line being formatted.

> The JVM config file, ``etc/jvm.config``, contains a list of command line

Reviewer: ```suggestion
The JVM configuration file, ``etc/jvm.config``, contains a list of command line
```

options used for launching the Java Virtual Machine.

The following provides an example of ``etc/jvm.config``.

.. code-block:: none

    -ea
    -XX:+UseG1GC
    -XX:G1HeapRegionSize=32M
    -XX:+UseGCOverheadLimit
    -XX:+ExplicitGCInvokesConcurrent
    -Xmx12G
    --add-opens=java.base/java.nio=org.apache.arrow.memory.core,ALL-UNNAMED

.. _config_properties:

> Config Properties

Reviewer: ```suggestion
Configuration Properties
```

> ^^^^^^^^^^^^^^^^^

Reviewer: Please edit this line to match the number of characters in the line being formatted.

> The config properties file, ``etc/flightshim.properties``, contains the

Reviewer: ```suggestion
The configuration properties file ``etc/flightshim.properties`` contains the
```

configuration for the Presto FlightShim server.

The following provides an example of ``etc/flightshim.properties``.

.. code-block:: none

    flight-shim.server=localhost
    flight-shim.server.port=9999
    flight-shim.server-ssl-certificate-file=/path/to/server.crt
    flight-shim.server-ssl-key-file=/path/to/server.key

.. _authentication:

Authentication

> ------------------

Reviewer: ```suggestion
--------------
```
